### PR TITLE
Address problems with lease warnings and over-gossiping configs.

### PR DESCRIPTION
**Pull request by spencerkimball**

### CODE REVIEW

File: gossip/server.go [hunk 1]
```diff
@@ -104,8 +104,14 @@ func (s *server) Gossip(args *proto.GossipRequest, reply *proto.GossipResponse)
 		if err := gob.NewDecoder(bytes.NewBuffer(args.Delta)).Decode(delta); err != nil {
 			return util.Errorf("infostore could not be decoded: %s", err)
 		}
-		if log.V(1) {
-			log.Infof("received delta infostore from client %s: %s", addr, delta)
+		if delta.infoCount() > 0 {
+			if delta.infoCount() > 0 {
```
Comment (bdarnell) [inline gossip/server.go]: Remove one of these lines.

File: storage/range.go [hunk 2]
```diff
@@ -113,7 +114,7 @@ type configDescriptor struct {
 
 // configDescriptors is a slice containing the accounting, permissions
```
Comment (bdarnell) [inline storage/range.go]: It's now an array, not a slice. (Why did you make this change, btw?)

File: storage/range.go [hunk 3]
```diff
@@ -895,7 +884,7 @@ func (r *Range) getLeaseForGossip() (bool, error) {
 		switch e := err.(type) {
 		// NotLeaderError means there is an active lease, leaseRejectedError
 		// means we tried to get one but someone beat us to it.
-		case *proto.NotLeaderError, *leaseRejectedError:
+		case *proto.NotLeaderError: //, *proto.LeaseRejectedError:
```
Comment (bdarnell) [inline storage/range.go]: Remove this or uncomment it.

File: storage/range.go [hunk 4]
```diff
@@ -915,6 +904,7 @@ func (r *Range) maybeGossipFirstRange() error {
 	if ok, err := r.getLeaseForGossip(); !ok || err != nil {
 		return err
 	}
+	log.Infof("gossiping first range from store %d, range %d", r.rm.StoreID(), r.Desc().RaftID)
```
Comment (bdarnell) [inline storage/range.go]: Do you want to keep this at Infof or make it a V(1)?

File: storage/range.go [hunk 5]
```diff
@@ -964,24 +956,28 @@ func (r *Range) maybeGossipConfigs(match func(configPrefix proto.Key) bool) {
...
+// instantiates/returns a config map and its checksum. Prefix
+// configuration maps include accounting, permissions, and zones.
+func (r *Range) loadConfigMap(keyPrefix proto.Key, configI interface{}) (PrefixConfigMap, uint32, error) {
 	kvs, err := engine.MVCCScan(r.rm.Engine(), keyPrefix, keyPrefix.PrefixEnd(), 0, proto.MaxTimestamp, true, nil)
 	if err != nil {
-		return nil, err
+		return nil, 0, err
 	}
 	var configs []*PrefixConfig
+	c := encoding.NewCRC32Checksum([]byte(nil))
```
Comment (bdarnell) [inline storage/range.go]: CRC32 is pretty weak and I'm worried about false matches here causing config changes to fail to propagate. This is far from performance-critical so just use a strong hash like SHA256.

File: gossip/server.go (hunk 1 above)
Comment (spencerkimball) [inline gossip/server.go]: Oops. Done.

**Comment by bdarnell** [comment]
LGTM

File: storage/range.go (hunk 2 above)
Comment (spencerkimball) [inline storage/range.go]: Because I wanted a constant-sized array for the checksums each range keeps track of.

**Comment by tschottdorf** [comment]
LGTM mod @bdarnell's comments.

File: storage/range.go (hunk 3 above)
Comment (spencerkimball) [inline storage/range.go]: Ooops. Done.

File: storage/range.go (hunk 4 above)
Comment (spencerkimball) [inline storage/range.go]: Infof is what it was when it lived inside store.go. I think it's important enough and only being logged from one node in the cluster. I'd rather leave it in.

File: storage/range.go (hunk 5 above)
Comment (spencerkimball) [inline storage/range.go]: Moved to sha256

**Pull request merged**